materialchart/src/main/java/app/priceguard/materialchart/Chart.kt

```diff
@@ -684,7 +638,102 @@ class Chart @JvmOverloads constructor(
                 }
             }
         }
+    }
+
+    private fun drawGridLine(canvas: Canvas) {
+
+        // Get chart data & max/min value
+        val chartData = dataset?.data ?: return
+
+        val maxY = chartData.maxOf { it.y }
+        val minY = chartData.minOf { it.y }
+        val spaceY = getSpace(maxY, minY)
+
+        val graphSpaceStartY = calculateYAxisFirstAndLastTick().second
+        val graphSpaceEndY = calculateYAxisFirstAndLastTick().first
+
+        val graphHeight = graphSpaceEndY - graphSpaceStartY
+
+        // Calculate available axis space
+        val availableSpace: Dp =
+            Px(width.toFloat()).toDp(context) - xAxisMarginStart - xAxisMarginEnd
+
+        //Draw GridLines
+        dataset?.gridLines?.sortedBy { it.value * -1 }?.forEach { data ->
+            val lineHeight: Px =
+                Px(1 - (data.value - minY) / spaceY) * graphHeight + graphSpaceStartY
+            if (minY > data.value || data.value > maxY) {
+                return@forEach
+            }
+            val axisStartPointX: Px = xAxisMarginStart.toPx(context)
+            val axisEndPointX: Px = (xAxisMarginStart + availableSpace).toPx(context)
+
+            gridLinePaint.setGridLinePaint()
+
+            canvas.drawLine(
+                axisStartPointX.value,
+                lineHeight.value,
+                axisEndPointX.value,
+                lineHeight.value,
+                gridLinePaint
+            )
+
+            // Draw ticks & labels
+            val labelString = data.name
+            gridLinePaint.getTextBounds(labelString, 0, labelString.length, bounds)
+
+            val textWidth = Px(bounds.width().toFloat())
+            val textHeight = Px(bounds.height().toFloat())
+
+            val labelStartPointX: Px = axisEndPointX - textWidth
+            val labelStartPointY: Px = lineHeight - textHeight
+
+            gridLinePaint.setTickPaint()
+
+            canvas.drawText(data.name, labelStartPointX.value, labelStartPointY.value, gridLinePaint)
+        }
+    }
+
+    private fun getDifference(maxValue: Float, minValue: Float): Float {
+        return if (maxValue == minValue) {
+            maxValue
+        } else {
+            maxValue - minValue
+        }
+    }
+
+    private fun getAvailableLabelCount(availableLabelSpace: Float, spacing: Float): Int {
+        // Number of ticks
+        // ~ 150dp : 3 (max, min, 50%)
+        // ~ 250dp : 5 (max, min, 25%, 50%, 75%)
+        // 250dp ~ : Auto
+        return when {
+            availableLabelSpace <= 150F -> 3
+            availableLabelSpace <= 250F -> 5
+            else -> {
+                (availableLabelSpace / spacing).toInt()
+            }
+        }
+    }
 
+    private fun getNeededLabelCount(
+        availableLabels: Int,
+        availableLabelSpace: Float,
+        spacing: Float
```

Comment:
We should handle these in Dp since it can be confusing whether it is Px or Dp. `availableLabelSpace` and `actualSpacing` values are are Float values of Dp.

Reply:
Thanks for mention this. I will fix this soon.